src/memory/AmdFlash.cc

```diff
@@ -47,21 +47,28 @@ AmdFlash::AmdFlash(const std::string& name, const ValidatedChip& validatedChip,
 	: motherBoard(config.getMotherBoard())
 	, chip(validatedChip.chip)
 {
-	auto numSectors = chip.geometry.sectorCount;
-	assert(writeProtectSectors.size() <= numSectors);
+	assert(writeProtectSectors.size() <= chip.geometry.sectorCount);
+
+	sectors.reserve(chip.geometry.sectorCount);
+	for (size_t address = 0; const Region& region : chip.geometry.regions) {
+		for (size_t regionSector = 0; regionSector < region.count; ++regionSector, address += region.size) {
+			sectors.push_back({
+				.address = address,
+				.size = region.size,
+				.writeProtect = sectors.size() < writeProtectSectors.size() && writeProtectSectors[sectors.size()]
+			});
+		}
+	}
```

Comment:
I like how this is now the only remaining location where we have this non-trivial nested loop.